k8s: Updated LastUpdated after waiting for endpoint status

tgraf opened the pull request:
Ensure that the LastUpdated timestamp is using the time-stamp after waiting for
the policy revision wait channel to be closed or timing out.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/4461)
<!-- Reviewable:end -->

raybejjani reviewed (approved):
I'm not sure if this change will change the order of operations in the way you intend. Since `enforcementSuccessful` is only used in one place, it would, at best, be a few nanoseconds before the timestamp.

It might be more effective to make the 1370 if-else a switch and default to setting the LastUpdated stamp there. In fact, simply setting this timestamp on like 1383 for all cases might be the most correct? since those other two cases would also suffer from any timing issues you've seen.

tgraf committed:
k8s: Updated LastUpdated after waiting for endpoint status

Ensure that the LastUpdated timestamp is using the time-stamp after waiting for
the policy revision wait channel to be closed or timing out.

Signed-off-by: Thomas Graf <thomas@cilium.io>
